packages/library/src/rpc-request-coalescer-gating.ts

```diff
@@ -0,0 +1,22 @@
+interface IHasParams {
+    params: unknown;
+}
+interface IHasMethod {
+    method: string;
+}
+
+export function shouldCoalesceRequest(payload: unknown): payload is IHasMethod & IHasParams {
+    if (payload == null || typeof payload !== 'object' || Array.isArray(payload)) {
+        return false;
+    }
+    if (!('jsonrpc' in payload) || payload.jsonrpc !== '2.0' || !('method' in payload)) {
+        return false;
+    }
+    switch (payload.method) {
+        case 'sendTransaction':
```

Comment:
Let's coalesce `sendTransaction` too -- there's no reason to send the same exact transaction if it's already in-flight.